src/gleam/bit_array.gleam

```diff
@@ -193,3 +193,48 @@ fn do_inspect(input: BitArray, accumulator: String) -> String {
     _ -> accumulator
   }
 }
+
+/// Compare two bit arrays as sequences of bytes.
+///
+/// ## Examples
+///
+/// ```gleam
+/// compare(<<1>>, <<2>>)
+/// // -> Lt
+///
+/// compare(<<"AB":utf8>>, <<"AA":utf8>>)
+/// // -> Gt
+///
+/// compare(<<1, 2:size(2)>>, with: <<1, 2:size(2)>>)
+/// // -> Eq
+/// ```
+///
+/// Only supported on Erlang target for now.
+///
+pub fn compare(a: BitArray, with b: BitArray) -> order.Order {
+  do_compare(a, b)
+}
+
+@external(javascript, "../gleam_stdlib.mjs", "bit_array_compare")
+fn do_compare(a: BitArray, with b: BitArray) -> order.Order {
```

Comment:
How come this implementation isn't used for JavaScript?

Reply:
Js does not have `:bits` part in BitArray support, so I had to come up with something else.

Reply:
I am not sure that it will work on js, I would have to try it.